### ISSUE TITLE
Always JUCE Assertion failure in juce_linux_XWindowSystem.cpp:190

### ISSUE THREAD

**Issue by kmatheussen** [juce-framework/JUCE#979]
Ubuntu 20, JUCE v6.1.4 (2021-12-20)

```
cd JUCE-6.1.4/examples/DemoRunner/Builds/LinuxMakefile
./build/DemoRunner
JUCE v6.1.4
JUCE Assertion failure in juce_linux_XWindowSystem.cpp:190
ERROR: X returned BadWindow (invalid Window parameter) for operation X_GetProperty
ERROR: X returned BadWindow (invalid Window parameter) for operation X_ChangeWindowAttributes
```

I get this assertion with any program that opens a window. In debug mode, I have commented out the assertion, and things seems to work if I do that.

**Comment by jacobweber** [comment]
Seeing this too, using the [pamplejuce](https://github.com/sudara/pamplejuce) Github actions running on Ubuntu 20.04 with Juce 6.1.4. It tries to run pluginval:

```
Archive:  pluginval.zip
  inflating: pluginval               
X Error of failed request:  BadWindow (invalid Window parameter)
  Major opcode of failed request:  20 (X_GetProperty)
  Resource id in failed request:  0x0
  Serial number of failed request:  54
  Current serial number in output stream:  54
pluginval v0.3.0 - JUCE v6.1.0
Started validating: /path/to/my.vst3
Random seed: 0x4540d3
Validation started: 30 Dec 2021 8:54:52pm

Strictness level: 5
-----------------------------------------------------------------
Starting test: pluginval / Scan for known types: 
/path/to/my.vst3...
Error: Process completed with exit code 1.
```

 Downgrading to 6.1.2 fixes it.

**Comment by robbert-vdh** [comment]
This is a gdb log and backtrace for opening any JUCE 6.1.3+ VST3 plugin in any host on Arch Linux with Xorg 21.1.2 libX11 1.7.3.1, and the latest develop branch JUCE (so the JUCE assertion failure line number should still be up to date):

<details>
<summary>backtrace</summary>

```
>>> break _XDefaultError 
Breakpoint 5 at 0x7ffff7e6f990
>>> r
The program being debugged has been started already.
Start it from the beginning? (y or n) y
Starting program: /usr/lib/carla/carla-bridge-native vst3 /home/robbert/.vst3/Spectral\ Compressor.vst3 \(none\) 0
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
libjack.so.0 loaded successfully!
[New Thread 0x7ffff586d640 (LWP 31846)]
[New Thread 0x7ffff4fcc640 (LWP 31847)]
[New Thread 0x7ffff47cb640 (LWP 31848)]
[Thread 0x7ffff47cb640 (LWP 31848) exited]
[Thread 0x7ffff4fcc640 (LWP 31847) exited]
JUCE v6.1.4
[New Thread 0x7ffff4fcc640 (LWP 31849)]
JUCE Assertion failure in juce_linux_XWindowSystem.cpp:190

Thread 1 "carla-bridge-na" received signal SIGTRAP, Trace/breakpoint trap.
0x00007ffff7c378ca in __futex_abstimed_wait_common64 () from /usr/lib/libpthread.so.0
>>> c
Continuing.
[Switching to Thread 0x7ffff4fcc640 (LWP 31849)]

Thread 5 "carla-bridge-na" hit Breakpoint 5, 0x00007ffff7e6f990 in _XDefaultError () from /usr/lib/libX11.so.6
>>> bt
#0  0x00007ffff7e6f990 in _XDefaultError () at /usr/lib/libX11.so.6
#1  0x00007ffff7e6fb05 in _XError () at /usr/lib/libX11.so.6
#2  0x00007ffff7e6c7a8 in  () at /usr/lib/libX11.so.6
#3  0x00007ffff7e6da2d in _XReply () at /usr/lib/libX11.so.6
#4  0x00007ffff7e52de6 in XGetWindowProperty () at /usr/lib/libX11.so.6
#5  0x00007ffff378eded in juce::XWindowSystemUtilities::GetXProperty::GetXProperty(_XDisplay*, unsigned long, unsigned long, long, long, bool, unsigned long) (this=0x7ffff4fcb830, display=0x7fffec01a5a0, window=0, atom=433, offset=0, length=9223372036854775807, shouldDelete=false, requestedType=433) at /home/robbert/Documenten/projecten/spectral-compressor/build/_deps/juce-src/modules/juce_gui_basics/native/x11/juce_linux_XWindowSystem.cpp:169
#6  0x00007ffff378f530 in juce::XWindowSystemUtilities::XSettings::update() (this=0x7fffec028be0) at /home/robbert/Documenten/projecten/spectral-compressor/build/_deps/juce-src/modules/juce_gui_basics/native/x11/juce_linux_XWindowSystem.cpp:212
#7  0x00007ffff378ef5e in juce::XWindowSystemUtilities::XSettings::XSettings(_XDisplay*) (this=0x7fffec028be0, d=0x7fffec01a5a0) at /home/robbert/Documenten/projecten/spectral-compressor/build/_deps/juce-src/modules/juce_gui_basics/native/x11/juce_linux_XWindowSystem.cpp:191
#8  0x00007ffff38469ff in std::make_unique<juce::XWindowSystemUtilities::XSettings, _XDisplay*&>(_XDisplay*&) () at /usr/include/c++/11.1.0/bits/unique_ptr.h:962
#9  0x00007ffff3799784 in juce::XWindowSystem::initialiseXSettings() (this=0x7fffec005790) at /home/robbert/Documenten/projecten/spectral-compressor/build/_deps/juce-src/modules/juce_gui_basics/native/x11/juce_linux_XWindowSystem.cpp:3033
#10 0x00007ffff3799e75 in juce::XWindowSystem::initialiseXDisplay() (this=0x7fffec005790) at /home/robbert/Documenten/projecten/spectral-compressor/build/_deps/juce-src/modules/juce_gui_basics/native/x11/juce_linux_XWindowSystem.cpp:3124
#11 0x00007ffff3791d50 in juce::XWindowSystem::XWindowSystem() (this=0x7fffec005790) at /home/robbert/Documenten/projecten/spectral-compressor/build/_deps/juce-src/modules/juce_gui_basics/native/x11/juce_linux_XWindowSystem.cpp:1442
#12 0x00007ffff362e910 in juce::SingletonHolder<juce::XWindowSystem, juce::CriticalSection, false>::getWithoutChecking() (this=0x7ffff3fc7b80 <juce::XWindowSystem::singletonHolder>) at /home/robbert/Documenten/projecten/spectral-compressor/build/_deps/juce-src/modules/juce_core/memory/juce_Singleton.h:103
#13 0x00007ffff361fda3 in juce::SingletonHolder<juce::XWindowSystem, juce::CriticalSection, false>::get() (this=0x7ffff3fc7b80 <juce::XWindowSystem::singletonHolder>) at /home/robbert/Documenten/projecten/spectral-compressor/build/_deps/juce-src/modules/juce_core/memory/juce_Singleton.h:92
#14 0x00007ffff360bbb1 in juce::XWindowSystem::getInstance() () at /home/robbert/Documenten/projecten/spectral-compressor/build/_deps/juce-src/modules/juce_gui_basics/native/x11/juce_linux_XWindowSystem.h:250
#15 0x00007ffff360c4e6 in juce::MessageThread::start()::{lambda()#1}::operator()() const (__closure=0x555555991df8) at /home/robbert/Documenten/projecten/spectral-compressor/build/_deps/juce-src/modules/juce_audio_plugin_client/VST3/../utility/juce_LinuxMessageThread.h:64
#16 0x00007ffff3659f22 in std::__invoke_impl<void, juce::MessageThread::start()::{lambda()#1}>(std::__invoke_other, juce::MessageThread::start()::{lambda()#1}&&) (__f=...) at /usr/include/c++/11.1.0/bits/invoke.h:61
#17 0x00007ffff3659ed7 in std::__invoke<juce::MessageThread::start()::{lambda()#1}>(juce::MessageThread::start()::{lambda()#1}&&) (__fn=...) at /usr/include/c++/11.1.0/bits/invoke.h:96
#18 0x00007ffff3659e84 in std::thread::_Invoker<std::tuple<juce::MessageThread::start()::{lambda()#1}> >::_M_invoke<0ul>(std::_Index_tuple<0ul>) (this=0x555555991df8) at /usr/include/c++/11.1.0/bits/std_thread.h:253
#19 0x00007ffff3659e58 in std::thread::_Invoker<std::tuple<juce::MessageThread::start()::{lambda()#1}> >::operator()() (this=0x555555991df8) at /usr/include/c++/11.1.0/bits/std_thread.h:260
#20 0x00007ffff3659e3c in std::thread::_State_impl<std::thread::_Invoker<std::tuple<juce::MessageThread::start()::{lambda()#1}> > >::_M_run() (this=0x555555991df0) at /usr/include/c++/11.1.0/bits/std_thread.h:211
#21 0x00007ffff79e23c4 in std::execute_native_thread_routine(void*) (__p=0x555555991df0) at /build/gcc/src/gcc/libstdc++-v3/src/c++11/thread.cc:82
#22 0x00007ffff7c2b259 in start_thread () at /usr/lib/libpthread.so.0
#23 0x00007ffff76e05e3 in clone () at /usr/lib/libc.so.6
>>>
```
</details>

**Comment by reuk** [comment]
On 6.1.3, I don't see this with XFCE/Unity, but do see it with i3. The assertion seems to be caused because the window manager doesn't provide a settings window.

The issue was fixed in 4f9a9c7bfc6dff0493c7e6cdc43b99cea1427777. Please update to develop and let us know if you still see the issue.
